TRENZ.Lib.RazorMail.SampleWebApi/Controllers/MailController.cs

```diff
@@ -1,68 +1,62 @@
 using Microsoft.AspNetCore.Mvc;
 
+using TRENZ.Lib.RazorMail.Interfaces;
 using TRENZ.Lib.RazorMail.Models;
 using TRENZ.Lib.RazorMail.SampleWebApi.Models;
-using TRENZ.Lib.RazorMail.Services;
 
 namespace TRENZ.Lib.RazorMail.SampleWebApi.Controllers;
 
 [Route("[controller]/[action]")]
 public class MailController(
-    IRazorEmailRenderer emailRenderer,
-    IConfiguration configuration,
-    ILoggerFactory loggerFactory
+    IMailRenderer emailRenderer
 )
     : ControllerBase
 {
-    private IRazorEmailRenderer EmailRenderer { get; } = emailRenderer;
-    private SmtpAccount SmtpAccount { get; } =
-        configuration.GetSection("SmtpAccount").Get<SmtpAccount>()!;
-
     [HttpPost]
-    public async Task<IActionResult> SendWithSystemNet([FromBody] SendSampleMailRequest request)
+    public async Task<IActionResult> SendWithSystemNet([FromBody] SendSampleMailRequest request, [FromKeyedServices("System.Net.Mail")] IMailClient client)
     {
         var renderedMail = await MakeRenderedMail(request);
 
-        var mail = new SystemNetMailSender(
-            from: request.From,
-            to: request.To.Select(x => new MailAddress(x)),
-            cc: [],
-            bcc: [],
-            replyTo: [],
-            renderedMail: renderedMail,
-            logger: loggerFactory.CreateLogger<SystemNetMailSender>()
-        );
+        var message = new MailMessage
```

Comment:
can't all of this be abstracted now since the concrete impl comes from the param?

Reply:
Yes